doc/source/user_guide/copy_on_write.rst

```diff
@@ -0,0 +1,213 @@
+
+The following example will operate inplace with CoW:
+
+.. ipython:: python
+
+    df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+    df.iloc[0, 0] = 100
+    df
+
+The object ``df`` does not share any data with any other object and hence no
+copy is triggered when updating the values. In contrast, the following operation
+triggers a copy of the data under CoW:
+
+
+.. ipython:: python
+
+    df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+    df2 = df.reset_index(drop=True)
+    df2.iloc[0, 0] = 100
+
+    df
+    df2
+
+``reset_index`` returns a lazy copy with CoW while it copies the data without CoW.
+Since both objects, ``df`` and ``df2`` share the same data, a copy is triggered
+when modifying ``df2``. The object ``df`` still has the same values as initially
+while ``df2`` was modified.
+
+If the object ``df`` isn't needed anymore after performing the ``reset_index`` operation,
+you can emulate an inplace-like operation through assigning the output of ``reset_index``
+to the same variable:
+
+.. ipython:: python
+
+    df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+    df = df.reset_index(drop=True)
+    df.iloc[0, 0] = 100
+    df
+
+The initial object gets out of scope as soon as the result of ``reset_index`` is
+reassigned and hence ``df`` does not share data with any other object. No copy
+is necessary when modifying the object. This is generally true for all methods
+listed in :ref:`Copy-on-Write optimizations <copy_on_write.optimizations>`.
+
+Previously, when operating on views, the view and the parent object was modified:
+
+.. ipython:: python
+
+    with pd.option_context("mode.copy_on_write", False):
+        df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+        view = df[:]
+        df.iloc[0, 0] = 100
+
+        df
+        view
+
+CoW triggers a copy when ``df`` is changed to avoid mutating ``view`` as well:
+
+.. ipython:: python
+
+    df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+    view = df[:]
+    df.iloc[0, 0] = 100
+
+    df
+    view
+
+Chained Assignment
+------------------
+
+Chained assignment references a technique where an object is updated through
+two subsequent indexing operations, e.g.
+
+.. ipython:: python
+
+    with pd.option_context("mode.copy_on_write", False):
+        df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+        df["foo"][df["bar"] > 5] = 100
+        df
+
+The column ``foo`` is updated where the column ``bar`` is greater than 5.
+This violates the CoW principles though, because it would have to modify the
+view ``df["foo"]`` and ``df`` in one step. Hence, chained assignment will
+consistently never work and raise a ``ChainedAssignmentError`` with CoW enabled:
+
+.. ipython:: python
+    :okexcept:
+
+    df = pd.DataFrame({"foo": [1, 2, 3], "bar": [4, 5, 6]})
+    df["foo"][df["bar"] > 5] = 100
+
+With copy on write this can either be done by using ``loc`` or doing this
+in multiple steps.
```

Comment:
"doing this in multiple steps" actually will also never work? 
At least if you mean doing the above literally in multiple steps like:

```
sub = df["foo"]
sub[df["bar"] > 5] = 100
```

(which is how I would interpret it)

Maybe showing the actual code for loc (`df.loc[df["bar"] > 5, "foo"] = 100`) could also help.
```

Reply:
I'll add the loc part.

But this would work in the sense of CoW? Updating sub but not df?

Reply:
Ah I get what you are referring to. I removed the comment.